tokio-console/src/state/mod.rs

```diff
@@ -538,3 +538,72 @@ fn pb_duration(dur: prost_types::Duration) -> Duration {
     let nanos = u64::try_from(dur.nanos).expect("duration should not be negative!");
     Duration::from_secs(secs) + Duration::from_nanos(nanos)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // This test should be run on all platforms. The console can display instrumentation data
+    // from different console subscribers that may be running on different operating systems.
+    // For instance, the console could be running on Windows, while the application is running on Linux.
+    // Therefore, it's important to ensure that paths, which can differ between operating systems,
+    // are displayed correctly in the console.
+    #[test]
+    fn test_format_location() {
+        // Linux style paths.
+        let location1 = proto::Location {
+            file: Some(
+                "/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.0.1/src/lib.rs"
+                    .to_string(),
+            ),
+            ..Default::default()
+        };
+        let location2 = proto::Location {
+            file: Some("/home/user/.cargo/git/checkouts/tokio-1.0.1/src/lib.rs".to_string()),
+            ..Default::default()
+        };
+        let location3 = proto::Location {
+            file: Some("/home/user/projects/tokio-1.0.1/src/lib.rs".to_string()),
+            ..Default::default()
+        };
+
+        assert_eq!(
+            format_location(Some(location1)),
+            "<cargo>/tokio-1.0.1/src/lib.rs"
+        );
+        assert_eq!(
+            format_location(Some(location2)),
+            "<cargo>/tokio-1.0.1/src/lib.rs"
+        );
+        assert_eq!(
+            format_location(Some(location3)),
+            "/home/user/projects/tokio-1.0.1/src/lib.rs"
+        );
+
+        // macOS style paths.
```

Comment:
this isn't a big deal, but my preference would still be to have separate tests for the macOS and Linux paths — that way, if one test fails, the test output will still indicate whether the other succeeded, which is potentially useful for determining if a bug effects all path styles, or just one OS. 
```suggestion
    }

    #[test]
    fn test_format_location_macos() {
        // macOS style paths.
```

Reply:
Updated.